vector/src/main/java/im/vector/app/core/utils/ExternalApplicationsUtil.kt

```diff
@@ -305,29 +305,21 @@ fun shareMedia(context: Context, file: File, mediaMimeType: String?) {
         return
     }
 
-    val sendIntent = ShareCompat.IntentBuilder(context)
+    val chooserIntent = ShareCompat.IntentBuilder(context)
             .setType(mediaMimeType)
             .setStream(mediaUri)
-            .getIntent()
-
-    sendShareIntent(context, sendIntent)
+            .setChooserTitle(R.string.share)
+            .createChooserIntent()
+    context.startActivity(chooserIntent)
 }
 
 fun shareText(context: Context, text: String) {
-    val sendIntent = Intent()
-    sendIntent.action = Intent.ACTION_SEND
-    sendIntent.type = "text/plain"
-    sendIntent.putExtra(Intent.EXTRA_TEXT, text)
-
-    sendShareIntent(context, sendIntent)
-}
-
-private fun sendShareIntent(context: Context, intent: Intent) {
-    try {
-        context.startActivity(Intent.createChooser(intent, context.getString(R.string.share)))
-    } catch (activityNotFoundException: ActivityNotFoundException) {
-        context.toast(R.string.error_no_external_application_found)
-    }
+    val chooserIntent = ShareCompat.IntentBuilder(context)
+            .setType("text/plain")
+            .setText(text)
+            .setChooserTitle(R.string.share)
+            .createChooserIntent()
+    context.startActivity(chooserIntent)
```

Comment:
Same remark. So maybe keep `fun sendShareIntent`, just modify it to remove the `createChooser` call.

Reply:
:heavy_check_mark: Changed in 74363ff8235764c0dc15ebb6b3dce42a6f52b56d.